src/cockpit/channels/filesystem.py

```diff
@@ -168,14 +168,26 @@ def delete(self, path: str, tag: 'str | None') -> str:
             os.unlink(path)
         return '-'
 
-    async def set_contents(self, path: str, tag: 'str | None', data: 'bytes | None') -> str:
+    async def set_contents(self, path: str, tag: 'str | None', data: 'bytes | None', size: 'int | None') -> str:
         dirname, basename = os.path.split(path)
         fd, tmpname = tempfile.mkstemp(dir=dirname, prefix=f'.{basename}-')
         try:
+            if size is not None:
+                logger.debug('fallocate(%s.tmp, %d)', path, size)
+                if size:  # posix_fallocate() of 0 bytes is EINVAL
+                    await self.in_thread(os.posix_fallocate, fd, 0, size)
```

Comment:
While at it, does it also make sense to add `posix_fadvise(os.POSIX_FADV_DONTNEED)` to avoid cache trashing? (This can be a follow-up of course)

Reply:
I'm not sure we know that as a general rule — it may well be the case that the user wants to use this immediately after uploading it.